cnxarchive/views/exports.py

```diff
@@ -85,55 +86,83 @@ def get_export(request):
     return resp
 
 
-def get_export_file(cursor, id, version, type, exports_dirs):
-    """Retrieve file associated with document."""
+def get_export_files(cursor, id, version, types, exports_dirs, read_file=True):
+    """Retrieve files associated with document."""
     request = get_current_request()
     type_info = dict(request.registry.settings['_type_info'])
-
-    if type not in type_info:
-        raise ExportError("invalid type '{}' requested.".format(type))
-
     metadata = get_content_metadata(id, version, cursor)
-    file_extension = type_info[type]['file_extension']
-    mimetype = type_info[type]['mimetype']
-    filename = '{}@{}.{}'.format(id, version, file_extension)
     legacy_id = metadata['legacy_id']
     legacy_version = metadata['legacy_version']
-    legacy_filenames = [
-        '{}-{}.{}'.format(legacy_id, legacy_version, ext)
-        for ext in LEGACY_EXTENSION_MAP[file_extension]
-    ]
-    slugify_title_filename = u'{}-{}.{}'.format(slugify(metadata['title']),
-                                                version, file_extension)
-
-    for dir in exports_dirs:
-        if safe_stat(dir):
+
+    reachable_dirs = [dir for dir in exports_dirs if safe_stat(dir)]
+
+    # 1 result per type, in the same order as the given types
+    results = []
+
+    for type in list(types):
```

Comment:
Hmm, why the extra wrap of `list()` around `types`, which is already a list? Harmless, but slightly jarring.